src/CloudFront.ts

```diff
@@ -0,0 +1,73 @@
+import { Credentials, CloudFront as AwsCloudFront } from 'aws-sdk'
+import { IamRole, IAM } from './IAM'
+import { S3, S3Bucket } from './S3';
+import { expect } from 'chai'
+import { isMatch } from 'lodash'
+
+class CloudFrontDistribution {
+    domainName;
+    distributionId;
+    role;
+    cloudFrontParams;
+    lazyAWSCloudFrontClient: AwsCloudFront
+    
+
+    constructor(distributionId: string, role?: IamRole) {
+        this.distributionId = distributionId;
+        this.role = role;
+        this.cloudFrontParams = { Id: this.distributionId };
+    }
+
+    toString() {
+        return `Cloud Front Distribution: ${this.distributionId}`
+    }
+
+    async cloudFrontClient() {
+        if (this.lazyAWSCloudFrontClient) {
+            return this.lazyAWSCloudFrontClient;
+        } else {
+            let temporaryCreds = this.role ? await this.role.credentials() : undefined
+            this.lazyAWSCloudFrontClient = new AwsCloudFront({ params: this.cloudFrontParams, credentials: temporaryCreds })
+            return this.lazyAWSCloudFrontClient;
+        }
+    }
+    async shouldExist() {
+        let cloudFrontClient = (await this.cloudFrontClient());
+        try {
+            let response = await cloudFrontClient.getDistribution(this.cloudFrontParams).promise()
+            expect(response).not.to.be.undefined;
+            expect(response.Distribution).not.to.be.undefined;
+            return true;
+        } catch (error ) { 
+            return false;

```

Comment:
Check for the expected error message in this situation, to avoid false positives.

Example:
see: https://github.com/MechanicalRock/beady-eye/blob/master/src/S3.ts

> ```
> async hasEncryptionByDefault() {
>         let s3 = (await this.s3Client())
>         try {
>             let response = await s3.getBucketEncryption(this.bucketParams).promise()
>             expect(response).not.to.be.undefined
>             return true
>         } catch (err) {
>             expect(err.code).to.equal('ServerSideEncryptionConfigurationNotFoundError')
>             return false
>         }
> }
> ```

Checking the error message returned prevents false positives.  

For example, if you don't have permission to view CloudFront, then this method would return `false`, if the distribution exists - which is incorrect behaviour.